test-suite/Math/NumberTheory/GCDTests.hs

```diff
@@ -86,6 +86,14 @@ toListReturnsCorrectValues :: Assertion
 toListReturnsCorrectValues =
   assertEqual "should be equal" (toList $ splitIntoCoprimes [(140, 1), (165, 1)]) [(5,2),(28,1),(33,1)]
 
+unionReturnsCorrectValues :: Assertion
+unionReturnsCorrectValues =
+  let a = splitIntoCoprimes [(700, 1), (165, 1)] -- [(5,3),(28,1),(33,1)]
+      b = splitIntoCoprimes [(360, 1), (210, 1)] -- [(2,4),(3,3),(5,2),(7,1)]
+      expected = [(2,4),(3,3),(5,3),(7,1),(28,1),(33,1)]
```

Comment:
Why is it expected? These numbers are not coprime with each other. I'd expect obtaining `[(2,6),(3,4),(5,5),(7,2),(11,1)]`.